File: day-08/api/src/main/java/com/bookstore/api/services/Abstract/AuthorService.java
package com.bookstore.api.services.Abstract;

import java.util.List;

import com.bookstore.api.entities.Author;
import com.bookstore.api.entities.models.ApiResponse;

public interface AuthorService {
    ApiResponse<List<Author>> getAllAuthors();

    ApiResponse<Author> getOneAuthor(int id);

    ApiResponse<Author> postOneAuthor(Author author);

    ApiResponse<Author> putOneAuthor(int id, Author author);

    void deleteOneAuthor(int id);
}

File: day-12/bs-store/src/services/AuthorService.js
import axios from "axios";

class AuthorService {
  constructor() {
    this.baseUrl = `${process.env.REACT_APP_BASE_ENDPOINT}/authors`;
  }

  async getAllAuthors() {
    return await axios
      .get(this.baseUrl)
      .then((resp) => resp.data)
      .catch((err) => console.log(err));
  }

  async getOneAuthor(id) {
    const url = `${this.baseUrl}/${id}`;
    return await axios
      .get(url)
      .then((resp) => resp.data)
      .catch((err) => console.log(err));
  }

  async postOneAuthor(payload) {
    return await axios
      .post(this.baseUrl, payload)
      .then((resp) => resp.data)
      .catch((err) => console.log(err));
  }

  async putOneAuthor(id, author) {
    return await axios
      .put(id, author)
      .then((resp) => resp.data)
      .catch((err) => console.log(err));
  }

  async deleteOneAuthor(id) {
    return await axios
      .delete(id)
      .then((resp) => resp)
      .catch((err) => console.log(err));
  }
}

export default AuthorService;
